File: gestao-angular/src/app/shared/model/endereco.ts
import {Pessoa} from "./pessoa";
import {Cidade} from "./cidade";
import {GridColumn} from "../utils/directives/grid.column.decorator";

export class Endereco {

  id: number;

  pessoa: Pessoa;

  // @GridColumn({ label: 'Cidade', type: 'texto', ordem: 5 })
  cidade: Cidade;

  @GridColumn({ label: 'Logradouro', type: 'texto', ordem: 1 })
  logradouro: string;

  @GridColumn({ label: 'Numero', type: 'texto',width:'100px', ordem: 2 })
  numero: string;

  @GridColumn({ label: 'Complemento', type: 'texto', ordem: 3 })
  complemento: string;

  @GridColumn({ label: 'Bairro', type: 'texto', ordem: 4 })
  bairro: string;

  @GridColumn({ label: 'CEP', type: 'texto', width:'150px' , ordem: 5 })
  cep: string;

  @GridColumn({ label: 'Ponto Ref.', type: 'texto', ordem: 6 })
  pontoReferencia: string;

  versao: number;
}0

File: gestao-spring/src/main/java/com/erp/gestao/model/endereco/Endereco.java
package com.erp.gestao.model.endereco;

import com.erp.gestao.model.Pessoa;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name ="endereco")
@Getter
@Setter
public class Endereco {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "pessoa")
    private Pessoa pessoa;

    @ManyToOne
    @JoinColumn(name = "cidade")
    private Cidade cidade;

    private String logradouro;

    private String numero;

    private String complemento;

    private String bairro;

    private String cep;

    private String pontoReferencia;

    @Version
    private Integer versao = 0;
}
